gammapy/catalog/hess.py

```diff
@@ -512,41 +514,87 @@ def spatial_model(self):
+                'linked' : each sub-component of a source is given as 
+                           a diffrent `SkyModel` but the spectral parameters
+                           except the mormalisation are linked.
+                'merged' : the sub-components are merged into a single `SkyModel`
+                           given as a `~gammapy.modeling.models.TemplateSpatialModel`
+                           with a `~gammapy.modeling.models.PowerLawNormSpectralModel`.
+                           In that case the relave weigths between the components
+                           cannot be adjusted.
         Returns
         -------
-        sky_model : `~gammapy.modeling.models.SkyModel`
-            Sky model of the catalog object.
+        sky_model : `~gammapy.modeling.models.Models`
+           Models of the catalog object.
         """
+
         spatial_type = self.data["Spatial_Model"]
         if spatial_type in {"2-Gaussian", "3-Gaussian"}:
             models = []
+            spectral_model = self.spectral_model(which=which)
             for component in self.components:
-                spectral_model = self.spectral_model(which=which)
+                spec_component = spectral_model.copy()
                 weight = component.data["Flux_Map"] / self.data["Flux_Map"]
-                spectral_model.parameters["amplitude"].value *= weight
+                spec_component.parameters["amplitude"].value *= weight
+                if components_status == "linked":
+                    for name in spec_component.parameters.names:
+                        if name not in ["norm", "amplitude"]:
+                            spec_component.__dict__[name] = spectral_model.parameters[
+                                name
+                            ]
                 model = SkyModel(
                     spatial_model=component.spatial_model(),
-                    spectral_model=spectral_model,
+                    spectral_model=spec_component,
                     name=component.name,
                 )
                 models.append(model)
 
-            return Models(models)
+            models = Models(models)
+            if components_status == "merged":
+                geom = self._get_components_geom(models)
+                return models.to_template_spatial_model(geom=geom, name=self.name)
+            else:
+                return models
         else:
             return SkyModel(
                 spatial_model=self.spatial_model(),
                 spectral_model=self.spectral_model(which=which),
                 name=self.name,
             )
 
+    @staticmethod
+    def _get_components_geom(models):
+        energy_axis = MapAxis.from_nodes(
+            10 ** np.arange(-0.95, 1.951, 0.1),
+            interp="log",
+            name="energy_true",
+            unit="TeV",
+        )
+        region = list_to_compound_region([m.spatial_model.to_region() for m in models])
```

Comment:
I think this can be reduced to:
```python3
regions = [c.spatial_model().evaluation_region for c in source.components]
geom = RegionGeom.from_regions(regions, binsz_wcs="0.02 deg")
wcs = geom.to_wcs_geom()
```